chalicelib/checks/wrangler_checks.py

```diff
@@ -472,3 +472,172 @@ def patch_file_size(connection, **kwargs):
     action.status = 'DONE'
     action.output = action_logs
     return action
+
+
+@check_function()
+def biosource_cell_line_value(connection, **kwargs):
+    '''
+    checks cell line biosources to make sure they have an associated ontology term
+    '''
+    check = init_check_res(connection, 'biosource_cell_line_value')
+
+    cell_line_types = ["primary cell", "primary cell line", "immortalized cell line",
+                       "in vitro differentiated cells", "induced pluripotent stem cell line",
+                       "stem cell", "stem cell derived cell line"]
+    biosources = ff_utils.search_metadata('search/?type=Biosource', ff_env=connection.ff_env, page_limit=200)
+    missing = []
+    for biosource in biosources:
+        # check if the biosource type is a cell/cell line
+        if biosource.get('biosource_type') and biosource.get('biosource_type') in cell_line_types:
+            # append if cell_line field is missing
+            if not biosource.get('cell_line'):
+                missing.append({'uuid': biosource['uuid'],
+                                '@id': biosource['@id'],
+                                'biosource_type': biosource.get('biosource_type'),
+                                'description': biosource.get('description'),
+                                'error': 'Missing cell_line metadata'})
+                # detail = 'In Biosource {}'.format(value['@id']) + \
+                #      ' - Missing Required cell_line field value for biosource type  ' + \
+                #      '{}'.format(bstype)
+    check.full_output = missing
+    check.brief_output = [item['uuid'] for item in missing]
+    if missing:
+        check.status = 'WARN'
+        check.summary = 'Cell line biosources found missing cell_line metadata'
+    else:
+        check.status = 'PASS'
+        check.summary = 'No cell line biosources are missing cell_line metadata'
+    return check
+
+
+@check_function()
+def external_expsets_without_pub(connection, **kwargs):
+    '''
+    checks external experiment sets to see if they are attributed to a publication
+    '''
+    check = init_check_res(connection, 'external_expsets_without_pub')
+
+    ext = ff_utils.search_metadata('search/?award.project=External&type=ExperimentSet',
```

Comment:
Same `frame=object` comment as above

Reply:
As an aside, `page_limit=200` makes some pretty big search requests that have to go through Fourfront. I would recommend setting that value lower, especially if you're getting embedded experiment sets